nidirect_taxonomy_navigator/src/Form/TaxonomyNavigatorForm.php

```diff
@@ -0,0 +1,265 @@
+          'group' => $group_class,
+        ],
+      ],
+    ];
+
+    if (!$can_reorder) {
+      unset($form['terms']['#tabledrag']);
+    }
+
+    // Build rows.
+    foreach ($terms as $key => $term) {
+      $form['terms'][$key]['#attributes']['class'][] = 'draggable';
+      $form['terms'][$key]['#attributes']['id'][] = $term->tid;
+      $form['terms'][$key]['#weight'] = $term->weight;
+      if ($highlight_tid == $term->tid) {
+        $form['terms'][$key]['#attributes']['style'][] = "background-color: lemonChiffon";
+      }
+
+      $form['terms'][$key]['name'] = [
+        '#title' => $term->name,
+        '#type' => 'link',
+        '#url' => Url::fromRoute('nidirect_taxonomy_navigator.taxonomy_navigator_form', [
+          'vocabulary' => $vocabulary->id(),
+          'taxonomy_term' => $term->tid,
+        ]),
+      ];
+
+      $form['terms'][$key]['weight'] = [
+        '#type' => 'weight',
+        '#title' => $this->t('Weight for @title', ['@title' => $term->name]),
+        '#title_display' => 'invisible',
+        '#default_value' => $term->weight,
+        '#attributes' => ['class' => [$group_class]],
+        '#disabled' => !$can_reorder,
+      ];
+
+      $form['terms'][$key]['tid'] = [
+        '#type' => 'hidden',
+        '#value' => $term->tid,
+      ];
+
+      $form['terms'][$key]['operations'] = [
+        '#type' => 'operations',
+        '#links' => [],
+      ];
+
+      $form['terms'][$key]['operations']['#links']['view'] = [
+        'title' => t('View'),
+        'url' => Url::fromRoute('entity.taxonomy_term.canonical', ['taxonomy_term' => $term->tid]),
+      ];
+
+      if ($can_edit) {
+        $form['terms'][$key]['operations']['#links']['edit'] = [
+          'title' => t('Edit'),
+          'url' => Url::fromRoute('entity.taxonomy_term.edit_form', ['taxonomy_term' => $term->tid], ['query' => $this->getDestinationArray()]),
+        ];
+
+        $form['terms'][$key]['operations']['#links']['set_parent'] = [
+          'title' => t('Set parent'),
+          'url' => Url::fromRoute('nidirect_taxonomy_navigator.set_parent_term_form', ['vocabulary' => $vocabulary->id(), 'taxonomy_term' => $term->tid], ['query' => $this->getDestinationArray()]),
+        ];
+      }
+
+      if ($can_delete) {
+        $form['terms'][$key]['operations']['#links']['delete'] = [
+          'title' => t('Delete'),
+          'url' => Url::fromRoute('entity.taxonomy_term.delete_form', ['taxonomy_term' => $term->tid], ['query' => $this->getDestinationArray()]),
+        ];
+      }
+    }
+
+    if ($can_reorder) {
+      $form['actions'] = ['#type' => 'actions'];
+      $form['actions']['submit'] = [
+        '#type' => 'submit',
+        '#value' => $this->t('Save'),
+      ];
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    foreach ($form_state->getValues() as $key => $value) {
+      // @TODO: Validate fields.
```

Comment:
What kind of extra fields need validating here? Is it needed at this stage?

Reply:
Bah, a hangover from Drupal generate